src/Compilers/CSharp/Portable/Symbols/Source/SourceMethodSymbolWithAttributes.cs

```diff
@@ -873,6 +886,15 @@ private void DecodeUnmanagedCallersOnlyAttribute(DecodeWellKnownAttributeArgumen
                 checkAndReportManagedTypes(param.Type, param.GetNonNullSyntaxNode(), isParam: true, arguments.Diagnostics);
             }
 
+            // In DecodeUnmanagedCallersOnlyAttribute, we check to see if this is a module initializer. In
+            // DecodeModuleInitializerAttribute, we check to see if this method is attributed with UnmanagedCallersOnly.
+            // When decoding a method we could have either order, so the first check will not find the presence of the
+            // other. The second check will find the presence and appropriately report an error.
```

Comment:
Can we check once, perhaps in `PostDecodeWellKnownAttributes()`?

Reply:
With the early attributes change, we can check just once in DecodeModuleInitializerAttribute

---
In reply to: [482613477](https://github.com/dotnet/roslyn/pull/47311#discussion_r482613477) [](ancestors = 482613477)